datasets/german_common_crawl/german_common_crawl.py

```diff
@@ -0,0 +1,143 @@
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="data_only", version=VERSION, description="Only the website text without metadata"
+        ),
+        datasets.BuilderConfig(
+            name="first_part",
+            version=VERSION,
+            description="Download only one part (2 GB) instead of everythong (200 GB)",
+        ),
+        datasets.BuilderConfig(name="metadata", version=VERSION, description="Metadata and raw text"),
+    ]
+
+    DEFAULT_CONFIG_NAME = "metadata"
+
+    def _info(self):
+        if self.config.name == "data_only":  # This is the name of the configuration selected in BUILDER_CONFIGS above
+            features = datasets.Features(
+                {
+                    "raw_content": datasets.Value("string"),
+                }
+            )
+        else:  # This is an example to show how to have different features for "first_domain" and "second_domain"
+            features = datasets.Features(
+                {
+                    "url": datasets.Value("string"),
+                    "date_download": datasets.Value("string"),
+                    "digest": datasets.Value("string"),
+                    "length": datasets.Value("int32"),
+                    "nlines": datasets.Value("int32"),
+                    "source_domain": datasets.Value("string"),
+                    "title": datasets.Value("string"),
+                    "raw_content": datasets.Value("string"),
+                    "cc_segment": datasets.Value("string"),
+                    "original_nlines": datasets.Value("int32"),
+                    "original_length": datasets.Value("int32"),
+                    "language": datasets.Value("string"),
+                    "language_score": datasets.Value("int32"),
+                    "perplexity": datasets.Value("int32"),
+                    "bucket": datasets.Value("string"),
+                }
+            )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+
+        if self.config.name == "first_part":
+            files = [dl_manager.download_and_extract(_URL[0])]
+
+        else:
+            files = []
+            for URL in _URL:
+                files.append(dl_manager.download_and_extract(URL))
```

Comment:
```suggestion
            files = dl_manager.download_and_extract(_URL)
```
this enables parallel downloads